crates/eframe/src/native/glow_integration.rs

```diff
@@ -691,10 +683,9 @@ impl GlowWinitRunning {
                 }
             }
 
-            winit::event::WindowEvent::ScaleFactorChanged { new_inner_size, .. } => {
-                if let Some(viewport_id) = viewport_id {
+            winit::event::WindowEvent::ScaleFactorChanged { .. } => {
+                if let Some(_viewport_id) = viewport_id {
                     repaint_asap = true;
-                    glutin.resize(viewport_id, **new_inner_size);
```

Comment:
In winit 0.29 `ScaleFactorChanged` does not expose the new OS suggested window size. But, a `Resized` event will come (recently fixed on macOS - https://github.com/rust-windowing/winit/pull/3214).